feat: initial backend setup with Node.js and Express

HijoDeHonor opened the pull request:
Set up initial backend configuration using Node.js and Express. Added basic structure and initial dependencies.

HijoDeHonor committed:
feat: initial backend setup with Node.js and Express

Set up initial backend configuration using Node.js and Express. Added basic structure and initial dependencies.

PiewDev reviewed (changes requested):
You must add a middleware to handle errors

HijoDeHonor committed:
feat:  add centralized error handler.

Added a centralized error handler to streamline error management. Refactored existing try-catch blocks to use the new error handler for improved code maintainability and readability.

HijoDeHonor committed:
refactor: organize error handlers into middlewares folder

Created a new folder named 'middlewares' and moved 'errorHandlers' into it for better project structure and organization.

PiewDev reviewed (changes requested):
In errorHandler always returns 400 code, we need to return other codes depending on the error. For default we use: 500 -internal server error, if error brings other code we use that:

```
 const statusCode = error.statusCode || 500; 
  const message = statusCode >= 500
    ? 'Internal Server Error' 
    : error.message;
```

HijoDeHonor committed:
refactor: update status code and message determination in error handler.

Modified the error handler to improve the logic for determining status codes and corresponding messages. This ensures more accurate and descriptive error responses.

HijoDeHonor committed:
fix: correct error in try-catch function.

Fixed a bug in the try-catch function to ensure proper error handling and execution flow.
